src/app/personside/infotabs/meldinger/traadvisning/verktoylinje/Verktoylinje.tsx

```diff
@@ -31,57 +34,99 @@ const OppgaveknapperStyle = styled.div`
     }
 `;
 
+const SvartLenkeKnapp = styled.button<{ aktiv?: boolean }>`
+    color: #3e3832;
+    position: relative;
+    border: none;
+    padding: 0;
+    border-radius: ${theme.borderRadius.knapp};
+    cursor: pointer;
+    background-color: transparent;
+    &:before {
+        content: '';
+        border-bottom: 1px solid transparent;
+        width: 100%;
+        left: 0;
+        bottom: 0;
+        position: absolute;
+    }
+    &:focus {
+        ${theme.focus}
+    }
+    &:hover {
+        &:before {
+            border-color: #3e3832;
+        }
+    }
+    ${props => {
+        if (props.aktiv === undefined) {
+            return '';
+        }
+        const direction = props.aktiv ? 'bottom' : 'top';
+        return `
+            padding-right: 1.25rem;
+            &:after {
+              content: '';
+              position: absolute;
+              height: 0.75rem;
+              width: 0.75rem;
+              display: block;
+              border: 0.5rem solid transparent;
+              border-${direction}-color: #3E3832;
+              right: 0;
+              ${direction}: 25%;
+            }
+        `;
+    }}
+`;
+
+enum FunksjonVindu {
+    JOURNALFORING,
+    OPPGAVE,
+    MERK
+}
+
 function Funksjoner(props: Props) {
-    const [visJournalforing, settVisJournalforing] = React.useState(false);
-    const [visOppgave, settVisOppgave] = React.useState(false);
-    const [visMerk, settVisMerk] = React.useState(false);
+    const [aktivtVindu, settAktivtVindu] = React.useState<FunksjonVindu | null>(null);
+    useEffect(() => {
```

Comment:
hvorfor trengs denne?

Reply:
Den er der for å lukke merk, oppgave og journalføringspanel når man bytter tråd.

Reply:
Men endret på dependencyListen fra `[props, settAktivtVindu]` til `[props.valgtTraad, settAktivtVindu]` for å gjøre det litt tydligere når effekten kjøres.

Reply:
Det er en liten svakhet ved useEffect synes jeg, at det ikke alltid er like åpenbart hva man ønsker å oppnå med en anonym funksjon